inference-engine/src/transformations/src/transformations/op_conversions/convert_gather_0d.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/convert_gather_0d.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertGather0D, "ConvertGather0D", 0);
+
+ngraph::pass::ConvertGather0D::ConvertGather0D() {
+    auto gather = ngraph::pattern::wrap_type<opset1::Gather>();
+
+    ngraph::matcher_pass_callback callback = [](pattern::Matcher &m) {
+        auto gather = std::dynamic_pointer_cast<ngraph::opset1::Gather>(m.get_match_root());
+        if (!gather) {
+            return false;
+        }
+
+        auto axes_constant = std::dynamic_pointer_cast<ngraph::opset1::Constant>(gather->input_value(2).get_node_shared_ptr());
+        if (!axes_constant) {
+            return false;
+        }
+        auto axis = axes_constant->cast_vector<int64_t>()[0];
+
+        // vector of new created nGraph operations
+        NodeVector new_ops;
+
+        // if the input with indices is scalar we need to unsqueeze it to 1D so plugins which do not support 0D can
+        // execute this layer. Then we need to squeeze the axis dimension to restore original shape of gather output
+        auto indices = gather->input_value(1);
+        const auto indices_rank = indices.get_partial_shape().rank();
+        if (indices_rank.is_dynamic()) {
+            return false;
+        }
+
+        bool squeeze_gather_output = indices_rank.get_length() == 0;
```

Comment:
Why you didn't put this condition under `if` ? :)

Reply:
Refactored